### PR TITLE
Add some stream features

### PR DESCRIPTION
**Pull request by chenquan**
like: https://github.com/chenquan/stream

### CODE REVIEW

**Comment by chenquan** [comment]
PTAL @kevwan

**Comment by chenquan** [comment]
PTAL @kevwan

**Comment by kevwan** [comment]
Some functions are provided already.
Like Split, and I'm not willing to add combined functions like ForeachOrdered.

**Comment by chenquan** [comment]
> Some functions are provided already.
> Like Split, and I'm not willing to add combined functions like ForeachOrdered.

1. Like Split? You mean SplitStream function? Maybe it should be retained, it can split the stream into different streams, can be used to speed up the operation
2. ForeachOrdered function will be removed,

**Comment by chenquan** [comment]
@kevwan Hello, do you have any other suggestions about the PR?

File: core/fx/stream.go [hunk 1]
```diff
@@ -63,6 +78,9 @@ func From(generate GenerateFunc) Stream {
 
 // Just converts the given arbitrary items to a Stream.
 func Just(items ...interface{}) Stream {
+	if len(items) == 0 {
```
Comment (kevwan) [inline core/fx/stream.go:81]: If no this code block, what's the problem?

File: core/fx/stream.go [hunk 2]
```diff
@@ -287,6 +310,35 @@ func (p Stream) Split(n int) Stream {
 	return Range(source)
 }
 
+// SplitSteam Returns a split Stream that contains multiple stream of chunk size n.
+func (p Stream) SplitSteam(n int) Stream {
```
Comment (kevwan) [inline core/fx/stream.go:314]: there is a Split function.

File: core/fx/stream.go (hunk 1 above)
Comment (chenquan) [reply core/fx/stream.go:81]: When the length of `items` is 0, we can simplify the operation and reuse `empty`.

File: core/fx/stream.go (hunk 2 above)
Comment (chenquan) [reply core/fx/stream.go]: thanks. I fixed it.

File: core/fx/stream.go (hunk 1 above)
Comment (kevwan) [reply core/fx/stream.go:81]: length 0 is a rare scenario, the only thing we need to do is working correctly.

File: core/fx/stream.go [hunk 3]
```diff
@@ -79,6 +97,11 @@ func Range(source <-chan interface{}) Stream {
 	}
 }
 
+// Concat Returns a concat Stream.
+func Concat(a Stream, others ...Stream) Stream {
+	return a.Concat(others...)
+}
```
Comment (kevwan) [inline core/fx/stream.go:103]: when we need to concat streams?

File: core/fx/stream.go [hunk 4]
```diff
@@ -380,6 +403,107 @@ func (p Stream) walkUnlimited(fn WalkFunc, option *rxOptions) Stream {
...
+			isFind = false
+			return
+		}
+	}
+	return
+}
+
+// Peek Returns a Stream consisting of the elements of this stream,
+// additionally performing the provided action on each element as elements are consumed from the resulting stream.
+func (p Stream) Peek(f ForEachFunc) Stream {
```
Comment (kevwan) [inline core/fx/stream.go:435]: Map works like a charm.

File: core/fx/stream.go (hunk 3 above)
Comment (chenquan) [reply core/fx/stream.go:103]: > when we need to concat streams?

When we want to do different filtering for two different streams, then do the same filtering, and finally output to the same file.

File: core/fx/stream.go (hunk 4 above)
Comment (chenquan) [reply core/fx/stream.go:435]: good. Less is more. I will delete it.

File: core/fx/stream.go (hunk 1 above)
Comment (chenquan) [reply core/fx/stream.go:81]: I will delete it.

File: core/fx/stream.go [hunk 5]
```diff
@@ -49,6 +49,21 @@ type (
 	}
 )
 
+// empty a empty Stream.
```
Comment (kevwan) [inline core/fx/stream.go:52]: this block should be deleted?

File: core/fx/stream.go [hunk 6]
```diff
@@ -380,6 +400,93 @@ func (p Stream) walkUnlimited(fn WalkFunc, option *rxOptions) Stream {
 	return Range(pipe)
 }
 
+// AnyMach Returns whether any elements of this stream match the provided predicate.
+// May not evaluate the predicate on all elements if not necessary for determining the result.
+// If the stream is empty then false is returned and the predicate is not evaluated.
+func (p Stream) AnyMach(f func(item interface{}) bool) (isFind bool) {
```
Comment (kevwan) [inline core/fx/stream.go:406]: MatchAny not AnyMatch

File: core/fx/stream.go [hunk 7]
```diff
@@ -380,6 +400,93 @@ func (p Stream) walkUnlimited(fn WalkFunc, option *rxOptions) Stream {
...
+	}
+	return
+}
+
+// Concat Returns a Stream that concat others streams
+func (p Stream) Concat(others ...Stream) Stream {
+	source := make(chan interface{})
+	wg := sync.WaitGroup{}
+	for _, other := range others {
+		if p == other {
```
Comment (kevwan) [inline core/fx/stream.go:435]: how to compare? undefined?

File: core/fx/stream.go (hunk 6 above)
Comment (chenquan) [reply core/fx/stream.go:406]: This naming method is referred to in Java stream.

File: core/fx/stream.go (hunk 5 above)
Comment (chenquan) [reply core/fx/stream.go:52]: ok,I will delete it.

File: core/fx/stream.go (hunk 7 above)
Comment (chenquan) [reply core/fx/stream.go:435]:
```go
a := Just(1,2,3)
a.Concat(a,a,a)
```
This will enable multiple goroutines.

**Pull request merged**